packages/@aws-cdk/custom-resource-handlers/lib/custom-resources-framework/classes.ts

```diff
@@ -215,9 +227,7 @@ export abstract class HandlerFrameworkClass extends ClassType {
         });
         getOrCreateMethod.addParameter({
           name: 'props',
-          type: scope.coreInternal
-            ? CORE_INTERNAL_CR_PROVIDER.CustomResourceProviderOptions
-            : CORE_MODULE.CustomResourceProviderOptions,
```

Comment:
This is one example of what the `ModuleImporter` class helps with. We can maintain one module defined, i.e., `CORE_MODULE`, with all of it's types, expressions, etc. rather than needing to define a duplicate module based on where the module would be imported from. Instead, when registering an import we can just register the same module with a different `fromLocation` specified. Now the code looks a lot cleaner.